rmgpy/molecule/fragment_utils.py

```diff
@@ -310,3 +313,321 @@ def merge_frag_list(to_be_merged):
+        options = add_starting_fragment_cut_if_needed(species_fragment, nearest_cutting_label, starting_fragment_smiles,species_cutting_threshold = species_cutting_threshold)
+        return options
+    else:
+        pibond_result = check_if_pibond_near_cutting_label(species_smiles)
+        if pibond_result!=False:
+            species_fragment, nearest_cutting_label = pibond_result
+            options = add_starting_fragment_cut_if_needed(species_fragment, nearest_cutting_label, starting_fragment_smiles,species_cutting_threshold = species_cutting_threshold)
+            return options
+        else:
+            return species_smiles
+
+def make_new_reaction_string(species_smiles, starting_fragment_smiles, frag_list):
+    frag_list_str = " + ".join(frag_list)
+    return f"{species_smiles} + {starting_fragment_smiles} => {frag_list_str}"
+
+
+def generate_add_partial_reattachment_reactions(seed_dir, starting_fragments):
+    seed_reactions_filename = os.path.join(seed_dir, "reactions.py")
+    seed_dictionary_filename = os.path.join(seed_dir, "dictionary.txt")
+
+    with open(seed_dictionary_filename,'r') as f:
+        dictionary_lines = f.readlines()
+
+    fragment_to_name_dictionary = {}
+
+    dlines_iter = iter(dictionary_lines)
+    name = next(dlines_iter,"end").strip().strip('\n')
+    line = next(dlines_iter,"end")
+    while line != "end" and name !="end":
+        species_adjlist =""
+        line = next(dlines_iter,"end")
+        while line.strip().strip('\n') != "":
+            species_adjlist +=line
+            line = next(dlines_iter, "end")
+
+        f = Fragment().from_adjacency_list(species_adjlist)
+        fragment_to_name_dictionary[f] = name
+        name = next(dlines_iter,"end").strip().strip('\n')
+
+    rxn_strs = []
+    for f in fragment_to_name_dictionary.keys():
+        smiles = f.smiles
+        try:
+            for starting_fragment in starting_fragments:
+                options = process_new_fragment(smiles, starting_fragment,species_cutting_threshold=12)
+                if type(options) == list:
+                    for option in options:
+                        option_frags = [Fragment().from_smiles_like_string(x) for x in option]
+                        if all([option in fragment_to_name_dictionary.keys() for option in option_frags]):
+                            smiles_name = fragment_to_name_dictionary[Fragment().from_smiles_like_string(smiles)]
+                            starting_fragment_name = fragment_to_name_dictionary[Fragment().from_smiles_like_string(starting_fragment)]
+                            option_name = [fragment_to_name_dictionary[x] for x in option_frags]
+                            rxn_str = make_new_reaction_string(smiles_name, starting_fragment_name, option_name)
+                            if rxn_str not in rxn_strs:
+                                rxn_strs.append(rxn_str)
+                    
+        except:
```

Comment:
Can we add explanation on when this is expected to fail and catch the error explicitly?

Reply:
Turns out the try: except: wasn't doing anything, so I removed it.